### ISSUE TITLE
padded label

### ISSUE THREAD

**Issue by CuongNguyen218** [DeNA/PyTorch_YOLOv3#44]
@hirose31 ,
I have a question. Why do you padded label with zeros by max_label in this line? I can't understand the purpose of lines and why you chosen max_label = 50
https://github.com/DeNA/PyTorch_YOLOv3/blob/cc008f0182e9d8048f6b5fb94e2938dced9e44fa/dataset/cocodataset.py#L119-L120

**Comment by CuongNguyen218** [comment]
and I have a question about training my own dataset. In #8  you said "Each element of 'labels' consists of [class, x1, y1, x2, y2] " but i know the labels of coco is [class x,y,w,h]. So which type of label i have to transform to train my own data with your repo

**Comment by CuongNguyen218** [comment]
![image](https://user-images.githubusercontent.com/43350150/69362287-d11fd580-0cc0-11ea-9a43-ea7c10c00aa9.png)
I have train yolov3 with your repo but i see loss at 2000s iter is so high. Did you see that in your training process. Can you provide me your pretrained model with COCO dataset

**Comment by hirotomusiker** [comment]
> Why do you padded label with zeros by max_label in this line?

It's for batch formulation. We have a trouble if the array size is not fixed at the output of the dataset class.

> training process

Your training log looks good to me. YOLOv3's loss is high because size_average=False, which is the same definition as darknet.

**Comment by hirotomusiker** [comment]
> In #8 you said "Each element of 'labels' consists of [class, x1, y1, x2, y2] " 

I was wrong at the issue and the docstring of label2yolobox has been corrected at https://github.com/DeNA/PyTorch_YOLOv3/pull/32.

Thank you!

**Comment by CuongNguyen218** [comment]
Thanks for your reply. I will retry and hope it work well.

**Comment by CuongNguyen218** [comment]
@hirotomusiker , Can you provide pretrained model on COCO dataset and test.py.

**Comment by hirotomusiker** [comment]
We can not provide our pretrained model due to license issues.
As for testing, you can evaluate your model by using train.py. see : https://github.com/DeNA/PyTorch_YOLOv3#evaluate-coco-ap

**Comment by CuongNguyen218** [comment]
@hirotomusiker 
Do you think batchsize can hurt your performance ?

**Comment by hirotomusiker** [comment]
Batchsize affects the final AP, which is the case for both our repo and the original darknet repo.
